File: golang/2760. 最长奇偶子数组.go
func longestAlternatingSubarray(nums []int, threshold int) int {
    res, dp := 0, 0
    for l := len(nums) - 1; l >= 0; l-- {
        if nums[l] > threshold {
            dp = 0
        } else if l == len(nums) - 1 || nums[l] % 2 != nums[l + 1] % 2 {
            dp++
        } else {
            dp = 1
        }
        if nums[l] % 2 == 0 && dp > res {
            res = dp
        }
    }
    return res
}

File: 刷题/cpp/2760. 最长奇偶子数组.cpp
class Solution {
public:
    int longestAlternatingSubarray(vector<int>& nums, int threshold) {
        bool flag=false;
        int ans=0,cnt=0;
        int l=0;
        for(int i=0;i<nums.size();i++){
            if(!flag){
                if(nums[i]%2 == 0 && nums[i]<=threshold){
                    flag=true;
                    cnt++;
                    l=i;
                }
            } else if(flag){
                if(i-1>=l && nums[i]%2!=nums[i-1]%2 && nums[i]<=threshold){
                    cnt++;
                    cout<<cnt<<endl;
                } else{
                    ans=max(ans,cnt);
                    if(nums[i]%2 != 0 || nums[i] > threshold){
                        flag=false;
                        cnt=0;
                    } else cnt=1;
                    
                }
            }
        }
        ans=max(ans,cnt);
        return ans;
    }
};